### PR TITLE
feature: shell completions

### PR DESCRIPTION
**Pull request by stevelr**
Signed-off-by: stevelr <steve@cosmonic.com>

### CODE REVIEW

**Comment by stevelr** [comment]
Todo:
 - [x] document where to install the generated files so they can be used.

File: src/completions.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    match generate_to(shell, &mut command, "wash", &output_dir) {
+        Ok(path) => {
+            let mut map = HashMap::new();
+            map.insert(
+                "path".to_string(),
+                path.to_string_lossy().to_string().into(),
+            );
+            Ok(CommandOutput::new(
+                format!("Generated completion file: {}", path.display()),
```
Comment (brooksmtownsend) [inline src/completions.rs]: What do I do with it though 😄

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
I think you have a TODO for this, just asking for more information on what to do with the completion file. It generated a file for me though 👍🏻

**Review by brooksmtownsend — COMMENTED** [review]
Worked great for me in zsh, but I couldn't get this to work in Warp or in PowerShell... @thomastaylor312 have you set up a PS profile before?

**Comment by thomastaylor312** [comment]
Yes, but I have to google it every time and it is never in the place I expect

**Comment by stevelr** [comment]
I had tested it in zsh, bash, and fish, but have not yet tested PowerShell.

File: Cargo.toml [hunk 2]
```diff
@@ -21,6 +21,7 @@ atelier_core = "0.2"
 bytes = "1.0"
 cargo_atelier = "0.2"
 clap = { version="3", features=["derive", "env"] }
+clap_complete = "3.2.4"
```
Comment (ricochet) [inline Cargo.toml:24]: Be sure to update this to the v4 version once #317 is merged.

**Comment by brooksmtownsend** [comment]
@stevelr would suggest that we either wrap this up once we can verify that it works in powershell, or close this issue.

**Comment by stevelr** [comment]
We should try to get it tested on PowerShell but I don't think that needs to be a blocker.

If we can't find someone to test PowerShell,  I think it's still worth releasing. It provides additive benefit for at least three other shells.

**Comment by brooksmtownsend** [comment]
@stevelr sounds good, I'll give it a shot on my Windows box later, if you're able to fix the conflicts I can 👍🏻 it

**Comment by vados-cosmonic** [comment]
Close in favor of https://github.com/wasmCloud/wash/pull/392 ?

**Comment by brooksmtownsend** [comment]
Closing in favor of #392

**Pull request closed without merging**